### ISSUE TITLE
A hyphen in environment name is ommited

### ISSUE THREAD

**Issue by r-divakaran** [cloudposse/terraform-null-label#47]
Hello,

I am using this module for naming all AWS resources created via terraform. Recently I wanted to use it for creating a secret manager resource as well. The template looks as below
```
module "secret_identifier" {
  source             = "git::https://github.com/cloudposse/terraform-null-label.git?ref=master"
  namespace          = "${var.namespace}"
  environment        = "${var.environment}"
  name               = "${var.secret_name}"
  attributes         = "${var.attributes}"
  delimiter          = "${var.delimiter}"
  tags               = "${var.tags}"
  additional_tag_map = "${var.additional_tags}"
  label_order        = ["namespace", "environment", "name", "attributes"]
}
```

I've set the delimiter to be `/`

Then am using it like
```
module "service_secret" {
  source               = "../modules/secret_manager_norotation"
  namespace            = "innovate"
  environment          = "dev-100}"
  attributes           = ["internal_dbconnection"]

  secret_name          = "internal_service"
  secret_description   = "Credentials set for internal service to access  db"
  user_name            = "bob_internal"

  tags                 = "${var.tags}"
  additional_tags      = "${map("service", "internal-service","purpose", "access_nternal_db")}"
}
```

Finally when its creating the secret its name is coming up like
**innovate/dev100/internalservice/internal_dbconnection**

So as you have noticed, it omitted the hyphens in environment and also omitted the underscore in the name. Is that an expected behaviour ? 
Any help will be greatly appreciated!

Thanks,
Raju.

**Comment by r-divakaran** [comment]
Can anyone please provide any inputs ?

**Comment by aknysh** [comment]
@r-divakaran 
thanks for testing and using the module.
looks like it's an expected behavior

https://github.com/cloudposse/terraform-null-label/blob/master/main.tf#L29
https://github.com/cloudposse/terraform-null-label/blob/master/main.tf#L37

it preserves just letters and digits.

If you open a PR to address the issue (change the Regex), we'll review promptly.

Thanks

**Comment by osterman** [comment]
Yes, we should make the regex (`"/[^a-zA-Z0-9]/"`) a variable.

**Comment by r-divakaran** [comment]
Thanks @aknysh for pointing that out.
I hope this is correct way to submit a pull request.
https://github.com/cloudposse/terraform-null-label/pull/48

**Comment by aknysh** [comment]
Addressed in #48, #49, #50
